core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -1412,6 +1412,11 @@ private QualifiedObjectName getRedirectedTableName(Session session, QualifiedObj
         requireNonNull(session, "session is null");
         requireNonNull(originalTableName, "originalTableName is null");
 
+        if (originalTableName.getCatalogName().isEmpty() || originalTableName.getSchemaName().isEmpty() || originalTableName.getObjectName().isEmpty()) {
+            // table cannot exist
```

Comment:
Should this be tested?

Reply:
this had started failing some other tests, added one in TestTableRedirection